.editorconfig

```diff
@@ -0,0 +1,25 @@
+# Documentation: https://EditorConfig.org
+# Inspired by Django .editorconfig file
+
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
+
+[*.{yaml,yml}]
+indent_size = 2
+
+[*.md]
+trim_trailing_whitespace = true
+indent_size = 4
```

Comment:
how does this work when doing two-line indenting inside yaml code blocks in the docs?

Reply:
It will still apply 4. I'll update `.md` files to use 2 by default. Honestly not that big of a problem since most markdown previews do not need the 4 spaces.

Reply:
no, that won't work either. 4 lines is required for some mkdocs features to work

Reply:
.editorconfig is not automatic in that sense. It only alters the default behavior of pressing "tab" in the editor. It is not going to modify files.

There are 2 options:
- Setting 4 then modifying to 2.
- Setting 2 and being aware of the 4 cases.

Let's go with whichever is more generic.

Reply:
Ok, go with 4